samples/sap-java-buildpack-tokenflow-usage/README.md

```diff
@@ -0,0 +1,147 @@
+	}
+
+	private JSONObject createJsonObjectFromVCAPServices() {
+		String vcapServices = System.getenv("VCAP_SERVICES");
+		return new JSONObject(vcapServices);
+	}
+
+	private void writeLine(HttpServletResponse response, String string) throws IOException {
+		response.getWriter().append(string);
+		response.getWriter().append("\n");
+	}
+
+}
+```
+
+# Deployment on Cloud Foundry
+To deploy the application, the following steps are required:
+- Configure the Application Router
+- Compile the Java application
+- Create a xsuaa service instance
+- Configure the manifest
+- Deploy the application
+git
+- Access the application
+
+## Configure the Application Router
+The [Application Router](./approuter/package.json) is used to provide a single entry point to a business application that consists of several different apps (microservices). It dispatches requests to backend microservices and acts as a reverse proxy. The rules that determine which request should be forwarded to which _destinations_ are called _routes_. The application router can be configured to authenticate the users and propagate the user information. Finally, the application router can serve static content.
+
+## Compile the Java application
+Run maven to package the application
+```shell
+mvn clean package
+```
+
+## Create the xsuaa service instance
+Use the [xs-security.json](./xs-security.json) to define the authentication settings and create a service instance
+```shell
+cf create-service xsuaa application xsuaa-tokenflow -c xs-security.json
+```
+
+## Configuration the manifest
+The [vars](../vars.yml) contains hosts and paths that need to be adopted.
+
+## Deploy the application
+Deploy the application using cf push. It will expect 1 GB of free memory quota.
+
+```shell
+cf push --vars-file ../vars.yml
+```
+
+## Cockpit administration tasks: Assign Role to your User
+Finally, as part of your Identity Provider, e.g. SAP ID Service, assign the deployed Role Collection(s) such as `Tokenflow_Buildpack_API_Viewer` to your user as depicted in the screenshot below and as documented [here](https://help.sap.com/viewer/65de2977205c403bbc107264b8eccf4b/Cloud/en-US/9e1bf57130ef466e8017eab298b40e5e.html).
+
+![](../images/SAP_CP_Cockpit_AssignRoleCollectionToUser.png)
+
+Further up-to-date information you can get on sap.help.com:
+- [Maintain Role Collections](https://help.sap.com/viewer/65de2977205c403bbc107264b8eccf4b/Cloud/en-US/d5f1612d8230448bb6c02a7d9c8ac0d1.html)
+- [Maintain Roles for Applications](https://help.sap.com/viewer/65de2977205c403bbc107264b8eccf4b/Cloud/en-US/7596a0bdab4649ac8a6f6721dc72db19.html).
+
+## Access the application
+After deployment, the application router will trigger authentication. If you have assigned the role-collection provided in the [xs-security.json](./xs-security.json) to your user, you will see an output like when calling `https://approuter-sap-java-buildpack-tokenflow-usage-<<ID>>.<<LANDSCAPE_APPS_DOMAIN>>`:
```

Comment:
take care that you do not access the service via approuter url https://approuter-sap-java-buildpack-tokenflow-usage ...